sugoi-api-core/src/main/java/fr/insee/sugoi/core/service/impl/OrganizationServiceImpl.java

```diff
@@ -63,7 +74,34 @@ public Organization findById(String realm, String storage, String id) {
         storage,
         SugoiEventTypeEnum.FIND_ORGANIZATION_BY_ID,
         Map.ofEntries(Map.entry("organizationId", id)));
-    return storeProvider.getReaderStore(realm, storage).getOrganization(id);
+    if (storage != null) {
+      try {
+
+        Organization org = storeProvider.getReaderStore(realm, storage).getOrganization(id);
+        org.addMetadatas("realm", realm.toLowerCase());
+        org.addMetadatas("userStorage", storage.toLowerCase());
+        return org;
+      } catch (Exception e) {
+        throw new EntityNotFoundException(
+            "Organization not found in realm " + realm + " and userStorage " + storage);
+      }
+    } else {
+      Realm r = realmProvider.load(realm);
+      for (UserStorage us : r.getUserStorages()) {
+        try {
+          Organization org = storeProvider.getReaderStore(realm, us.getName()).getOrganization(id);
+          if (org != null) {
+            org.addMetadatas("realm", realm);
+            org.addMetadatas("userStorage", us.getName());
+            return org;
+          }
+        } catch (Exception e) {
+          logger.info(
```

Comment:
why info ? Debug should be fine here (this is something expected to happen often.